src/Umbraco.Web/WebApi/Filters/AngularAntiForgeryHelper.cs

```diff
@@ -111,15 +107,41 @@ public static bool ValidateTokens(string cookieToken, string headerToken)
         /// <returns></returns>
         public static bool ValidateHeaders(HttpRequestHeaders requestHeaders, out string failedReason)
         {
-            var cookieToken = requestHeaders
-                .GetCookies()
-                .Select(c => c[CsrfValidationCookieName])
-                .FirstOrDefault();
+            var cookieToken = GetCookie(requestHeaders, CsrfValidationCookieName);
 
             return ValidateHeaders(
                 requestHeaders.ToDictionary(x => x.Key, x => x.Value).ToArray(),
-                cookieToken == null ? null : cookieToken.Value,
+                cookieToken == null ? null : cookieToken,
                 out failedReason);
         }
+
+        /// <summary>
+        /// Retrieves an individual cookie from the cookies collection
+        /// Adapted from: https://stackoverflow.com/a/29057304/5018
+        /// </summary>
+        /// <param name="requestHeaders"></param>
+        /// <param name="cookieName"></param>
+        /// <returns></returns>
+        private static string GetCookie(HttpRequestHeaders requestHeaders, string cookieName)
+        {
+            var cookieRequestHeader = requestHeaders.Where(h => h.Key.Equals("Cookie", StringComparison.InvariantCultureIgnoreCase)).ToArray();
+            if (cookieRequestHeader.Any() == false)
+                return null;
+
+            var cookiesHeader = cookieRequestHeader.FirstOrDefault();
+            if (cookiesHeader.Value == null)
+                return null;
+
+            var cookiesHeaderValue = cookiesHeader.Value.FirstOrDefault();
+            if (cookiesHeaderValue == null)
+                return null;
+
+            var cookieCollection = cookiesHeaderValue.Split(';');
+            var cookiePair = cookieCollection.FirstOrDefault(c => c.Trim().StartsWith(cookieName, StringComparison.InvariantCultureIgnoreCase));
```

Comment:
This check here seems a little odd too, we shouldn't be doing a `StartsWith` here because that can give misleading results if there are two cookie names prefixed with the same `cookieName` but are actually different. We need to re-split on the `=` symbol.  I'll update